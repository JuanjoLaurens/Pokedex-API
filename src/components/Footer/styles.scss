@import '../../scss/bootstrap';

.pd-footer {
  display: none !important;

  @media only screen and (max-width: map-get($map: $grid-breakpoints, $key: lg)) {
    display: inline-flex !important;
  }

  &__item {
    background-color: $white;

    &__link {
      outline: none !important;
      border: none !important;
      color: $text-muted;

      &-active {
        color: $primary !important;
      }
    }
  }
}
