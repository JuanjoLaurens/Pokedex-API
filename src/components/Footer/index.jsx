import cn from 'classnames/bind';
import React, { memo } from 'react';
import { Nav } from 'react-bootstrap';
import { Link } from 'react-router-dom';

import styles from './styles.scss';

import { paths } from '../../constants';
import { useLinks } from '../../hooks';

const cx = cn.bind(styles);

const Footer = () => {
  const { isActive } = useLinks();

  return (
    <Nav className={cx('pd-footer fixed-bottom shadow border-top')} fill variant="tabs">
      <Nav.Item className={cx('pd-footer__item')}>
        <Nav.Link
          active={isActive(paths.client.HOME)}
          as={Link}
          to={paths.client.HOME}
          className={cx('pd-footer__item__link display-inline-flex align-items-center', {
            'pd-footer__item__link-active': isActive(paths.client.HOME),
          })}
        >
          <i className="material-icons mt-1">home</i>
        </Nav.Link>
      </Nav.Item>
      <Nav.Item className={cx('pd-footer__item')}>
        <Nav.Link
          active={isActive(paths.client.FAVORITES)}
          as={Link}
          to={paths.client.FAVORITES}
          className={cx('pd-footer__item__link display-inline-flex align-items-center', {
            'pd-footer__item__link-active': isActive(paths.client.FAVORITES),
          })}
        >
          <i className="material-icons mt-1">favorite</i>
        </Nav.Link>
      </Nav.Item>
    </Nav>
  );
};

export default memo(Footer);
