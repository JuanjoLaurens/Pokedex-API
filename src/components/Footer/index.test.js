import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Footer from './index';

jest.mock('react-router-dom', () => ({
  useLocation: () => ({}),
}));

describe('Footer', () => {
  it('renders correctly according to snapshot', () => {
    const wrapper = shallow(<Footer />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
