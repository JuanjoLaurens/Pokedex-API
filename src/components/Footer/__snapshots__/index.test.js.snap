// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly according to snapshot 1`] = `
<Nav
  className="pd-footer fixed-bottom shadow border-top"
  fill={true}
  justify={false}
  variant="tabs"
>
  <NavItem
    className="pd-footer__item"
  >
    <NavLink
      active={false}
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      className="pd-footer__item__link display-inline-flex align-items-center"
      disabled={false}
      to="/"
    >
      <i
        className="material-icons mt-1"
      >
        home
      </i>
    </NavLink>
  </NavItem>
  <NavItem
    className="pd-footer__item"
  >
    <NavLink
      active={false}
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      className="pd-footer__item__link display-inline-flex align-items-center"
      disabled={false}
      to="/favorites"
    >
      <i
        className="material-icons mt-1"
      >
        favorite
      </i>
    </NavLink>
  </NavItem>
</Nav>
`;
